lucene/core/src/java/org/apache/lucene/search/TopFieldCollector.java

```diff
@@ -555,7 +519,7 @@ public static void populateScores(ScoreDoc[] topDocs, IndexSearcher searcher, Qu
 
   final void add(int slot, int doc) {
     bottom = pq.add(new Entry(slot, docBase + doc));
-    queueFull = totalHits == numHits;
+    queueFull = slot == numHits - 1;
```

Comment:
Hmm why was this needed?  Did we lose (easy) access to `totalHits`?

Can we `assert slot < numHits`?

Reply:
It's because in `PagingFieldCollector` we care about the number of hits collected since the first page, rather than `totalHits`, and this abstracts away the difference, since `slot` is either `totalHits-1` or `hitsCollected-1`

Reply:
maybe leave a comment about it there?

Reply:
Thanks, I did